### ISSUE TITLE
Add continuous integration

### ISSUE THREAD

**Issue by nicoulaj** [zsh-users/zsh-syntax-highlighting#440]
Goals: run test suite on master + other branches + pull requests, against released and development versions of zsh.

**Comment by nicoulaj** [comment]
Done thanks to @desyncr zsh docker containers :)

Remaining work: test against zsh development version. To do this I guess we can build on @desyncr's project (https://github.com/desyncr/zsh-docker) and improve it to also build zsh head periodically (can be done in Travis-CI: https://docs.travis-ci.com/user/cron-jobs).

**Comment by danielshahaf** [comment]
Thanks @nicoulaj!  I've made a couple of minor tweaks in followup revisions.

**Comment by desyncr** [comment]
@nicoulaj Adding `master` to `travis.yml` configuration will suffice (see [here](https://github.com/desyncr/zsh-docker/blob/master/.travis.yml#L15)). I'll update it later tonight!

`desyncr/zsh-docker` currently runs monthly by cronjob.

**Comment by nicoulaj** [comment]
@desyncr sorry, I got your message a little too late, I had already started on this. So I forked zsh-docker to https://github.com/zsh-users/zsh-docker.

The changes I made are:
 * Add zsh-master
 * Add all versions since zsh 4.3.5
 * Deploy to an organization instead of a personal account: https://hub.docker.com/u/zshusers
 * Additions to the Dockerfile (enable modules, build doc)
 * Suppressions in the Dockerfile: remove git and ssh config

I added you to both projects in case you want to contribute.

I have switched zsh-syntax-highlighting to these images, so now the tests are run daily against zsh master.
